phoenix-core/src/main/java/org/apache/phoenix/optimize/QueryOptimizer.java

```diff
@@ -312,6 +317,16 @@ private static QueryPlan addPlan(PhoenixStatement statement, SelectStatement sel
         return null;
     }
 
+    // returns true if we can still use the index
+    // retuns false if we've been in PENDING_DISABLE too long - index should be considered disabled
+    private static boolean isUnderPendingDisableThreshold(PhoenixStatement statement, PTable indexTable) {
+        return EnvironmentEdgeManager
+                .currentTimeMillis() - indexTable.getIndexDisableTimestamp() <= statement
```

Comment:
Since this is running client-side, the EnvironmentEdgeManager.currentTimeMillis() would be a client timestamp and the indexDisableTimestamp is a server-side timestamp so you can't reliably compare them directly. Instead, keep a bit more state client-side by storing the timestamp at which the index write failure occurred in the index TableRef (which isn't shared across multiple threads/connections). Then you can compare that timestamp here with the current time.

Reply:
I wanted to add a member variable to hold the threshold, but these methods are static, and I wasn't sure if adding a mutable static was advisable.  The call to isUnderPendingDIsableThreshold() is the last thing in the if-statement, so this only gets called in a PENDING_DISABLE scenario.  I'll add the optimization to use ReadOnlyProps, and use the client-side timestamp

Reply:
Sounds fine